### PR TITLE
AutologgingEventLogger: Pass all arguments to `log_autolog_called` in keyword form

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

This PR modifies `@autologging_integration` to pass `autolog()` arguments to `log_autolog_called` in keyword format to enable event loggers to more easily identify important configuration parameters (e.g., `disable`) without examining positional arguments. It also deprecates the `call_args` parameter for positional arguments in `log_autolog_called`.

## How is this patch tested?

Unit tests

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/autologging_utils.py [hunk 1]
```diff
@@ -600,8 +604,12 @@ def log_autolog_called(self, integration, call_args, call_kwargs):
         is invoked (e.g., when a user invokes `mlflow.sklearn.autolog()`)
 
         :param integration: The autologging integration for which `autolog()` was called.
-        :param config_args: The positional arguments passed to the `autolog()` call.
-        :param config_kwargs: The keyword arguments passed to the `autolog()` call.
+        :param call_args: **DEPRECATED** The positional arguments passed to the `autolog()` call.
+                          This field is empty in MLflow > 1.13.1; all arguments are passed in
+                          keyword form via `call_kwargs`.
```
Comment (harupy) [inline mlflow/utils/autologging_utils.py:609]: Should we raise a deprecation warnign if `call_args` is not an empty tuple?

Comment (dbczumar) [reply mlflow/utils/autologging_utils.py:609]: Added a warning!

Comment (harupy) [reply mlflow/utils/autologging_utils.py:609]: This might be an overkill but do we need a test?

**Review by harupy — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/3957/files#r553818173 has been addressed

File: tests/autologging/test_autologging_utils.py [hunk 2]
```diff
@@ -568,3 +575,57 @@ def autolog(disable=False):
...
+    AutologgingEventLogger.set_logger(AutologgingEventLogger())
+
+    class PatchObj:
+        def test_fn(self):
+            pass
+
+    # Test successful autologging workflow
+    AutologgingEventLogger.get_logger().log_autolog_called(
+        "test_integration", ("a"), {"b": 1, "c": "d"}
+    )
```
Comment (harupy) [inline tests/autologging/test_autologging_utils.py:590]: Can we use `pytest.raiess` here to ensure the deprecation warning is emitted?

File: mlflow/utils/autologging_utils.py (hunk 1 above)
Comment (dbczumar) [reply mlflow/utils/autologging_utils.py:609]: Done! Also added a case verifying that methods on the default logging impl don't throw.

File: tests/autologging/test_autologging_utils.py (hunk 2 above)
Comment (dbczumar) [reply tests/autologging/test_autologging_utils.py:590]: Added a separate test case for this.

Comment (harupy) [reply tests/autologging/test_autologging_utils.py:590]: I just saw https://github.com/mlflow/mlflow/pull/3957/files#diff-040921246593fba29d003ad475ba69ca9e9149f4df85d4aaf5f6d7c9407aaefaR634

File: tests/autologging/test_autologging_utils.py [hunk 3]
```diff
@@ -568,3 +575,70 @@ def autolog(disable=False):
...
+        (1000),
+        {"a": 2},
+        Exception("patch error"),
+    )
+
+
+def test_autologging_event_logger_default_impl_warns_for_log_autolog_called_with_deprecated_args():
+    AutologgingEventLogger.set_logger(AutologgingEventLogger())
+
+    with pytest.warns(DeprecationWarning):
```
Comment (harupy) [inline tests/autologging/test_autologging_utils.py:637]: Can we use a `match` arugment here 


```python
with pytest.warns(DeprecationWarning, "Received 1 positional arguments"):
```

**Review by harupy — APPROVED** [review]
Thanks for the quick fixes :) LGTM

**Pull request merged**